src/schemas/json/jj.json

```diff
@@ -0,0 +1,136 @@
+            "properties": {
+                "hostname": {
+                    "type": "string",
+                    "format": "hostname"
+                },
+                "username": {
+                    "type": "string"
+                },
+                "timestamp": {
+                    "type": "string",
+                    "format": "date-time"
+                }
+            }
+        },
+        "push": {
+            "type": "object",
+            "properties": {
+                "branch-prefix": {
+                    "type": "string",
+                    "description": "Prefix used when pushing a change ID as a new branch",
+                    "default": "push-"
+                }
+            }
+        },
+        "ui": {
+            "type": "object",
+            "description": "UI settings",
+            "properties": {
+                "allow-init-native": {
+                    "type": "boolean",
+                    "description": "Whether to allow initializing a repo with the native backend"
+                },
+                "relative-timestamps": {
+                    "type": "boolean",
+                    "description": "Whether to change timestamps to be rendered as a relative description instead of a full timestamp",
+                    "default": false
+                },
+                "default-revset": {
+                    "type": "string",
+                    "description": "Default set of revisions to show when no explicit revset is given for jj log and similar commands",
+                    "default": "@ | (remote_branches() | tags()).. | ((remote_branches() | tags())..)-"
+                },
+                "color": {
+                    "type": "string",
+                    "description": "Whether to colorize command output",
+                    "enum": ["always", "never", "auto"],
+                    "default": "auto"
+                },
+                "pager": {
+                    "type": "string",
+                    "description": "Pager to use for displaying command output",
+                    "default": "less -FRX"
+                },
+                "editor": {
+                    "type": "string",
+                    "description": "Editor to use for commands that involve editing text"
+                },
+                "diff-editor": {
+                    "type": "string",
+                    "description": "Editor tool to use for editing diffs",
+                    "default": "meld"
+                },
+                "merge-editor": {
+                    "type": "string",
+                    "description": "Tool to use for resolving three-way merges. Behavior for a given tool name can be configured in merge-tools.TOOL tables"
+                }
+            }
+        },
+        "colors": {
+            "type": "object",
+            "description": "Mapping from jj formatter labels to ANSI color names"
```

Comment:
nit: This makes it sound like the names are defined by ANSI, which I don't think is true

Reply:
Where can you find the full list of supported colors? And is there a list of the standard set of labels somewhere?

Reply:
The colors are defined [here](https://github.com/martinvonz/jj/blob/9974d67da249a69a0833f308b060248e37a47d1a/src/formatter.rs#L183).

The standard labels are spread out a bit. Some come [from the template parser](https://github.com/martinvonz/jj/blob/9974d67da249a69a0833f308b060248e37a47d1a/src/template_parser.rs#L286-L304), and some are defined through calls to `with_label()` throughout various files in [`src/`](https://github.com/martinvonz/jj/tree/main/src) (mainly `commands.rs`).

Reply:
Is it a fixed list of labels & colors? As a user, how would I best discover when I had an urge to customize some particular colors what their labels are called and what color names are supported?

Reply:
The list of colors is fixed, but the syntax will be extended in the future when we add support for background colors, underlining, etc.

Good question about the labels. Maybe we should add an option to produce XML- or JSON-formatted output. Then the user could find the possible labels there. We should probably also document the labels somewhere...

Reply:
Alright, eventually we'll need to figure out how to make the list of labels make sense, but I explicitly added the list of supported color names into the schema.